exonum/src/encoding/fields.rs

```diff
@@ -459,6 +460,56 @@ impl<'a> Field<'a> for SocketAddr {
     }
 }
 
+impl<'a> Field<'a> for ConnectInfo {
```

Comment:
Not using `encoding_struct` increases the amount of code that is hardly analyzable, potentially leads to UB, and will get thrown out relatively soon. I'd advise to use `encoding_struct` and implement address resolving on the level of `ConnectListConfig` if we can't decouple resolve from deserialization right now.